### PR TITLE
Add shellcheck and fix errors reported by shellcheck

### PR DESCRIPTION
**Pull request by umireon**
`install-guard.sh` contains some so-called bashisms and may not work under some POSIX sh.
This PR will make this project detect bashisms and fix them.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by umireon** [comment]
https://github.com/aws-cloudformation/cloudformation-guard/pull/169/checks?check_run_id=2850721254

```
In install-guard.sh line 20:
        while read MAJOR_VER; read VERSION; do
              ^--^ SC2162: read without -r will mangle backslashes.
                              ^--^ SC2162: read without -r will mangle backslashes.


In install-guard.sh line 21:
            mkdir -p ~/.guard/$MAJOR_VER ~/.guard/bin ||
                              ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
            mkdir -p ~/.guard/"$MAJOR_VER" ~/.guard/bin ||


In install-guard.sh line 24:
            download https://github.com/aws-cloudformation/cloudformation-guard/releases/download/$VERSION/cfn-guard-v$MAJOR_VER-$OS_TYPE-latest.tar.gz > /tmp/guard.tar.gz ||
                                                                                                  ^------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                                                                      ^--------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                                                                                 ^------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
            download https://github.com/aws-cloudformation/cloudformation-guard/releases/download/"$VERSION"/cfn-guard-v"$MAJOR_VER"-"$OS_TYPE"-latest.tar.gz > /tmp/guard.tar.gz ||


In install-guard.sh line 26:
            tar -C ~/.guard/$MAJOR_VER -xzf /tmp/guard.tar.gz ||
                            ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
            tar -C ~/.guard/"$MAJOR_VER" -xzf /tmp/guard.tar.gz ||


In install-guard.sh line 28:
            ln -sf ~/.guard/$MAJOR_VER/cfn-guard-v$MAJOR_VER-$OS_TYPE-latest/cfn-guard ~/.guard/bin ||
                            ^--------^ SC2086: Double quote to prevent globbing and word splitting.
                                                  ^--------^ SC2086: Double quote to prevent globbing and word splitting.
                                                             ^------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
            ln -sf ~/.guard/"$MAJOR_VER"/cfn-guard-v"$MAJOR_VER"-"$OS_TYPE"-latest/cfn-guard ~/.guard/bin ||


In install-guard.sh line 37:
    local _ostype="$(uname -s)"
    ^-----------^ SC2039: In POSIX sh, 'local' is undefined.
          ^-----^ SC2155: Declare and assign separately to avoid masking return values.

For more information:
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, 'local' is undefined.
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

**Review by dchakrav-github — APPROVED** [review]

**Review by priyap286 — APPROVED** [review]

File: install-guard.sh [hunk 1]
```diff
@@ -17,24 +17,24 @@ main() {
 
     get_os_type
     get_latest_release |
-        while read MAJOR_VER; read VERSION; do
-            mkdir -p ~/.guard/$MAJOR_VER ~/.guard/bin ||
+        while read -r MAJOR_VER; read -r VERSION; do
+            mkdir -p ~/".guard/$MAJOR_VER" ~/.guard/bin ||
```
Comment (priyap286) [inline install-guard.sh:21]: Can you tell me why you changed it to `~/".guard/$MAJOR_VER"` instead of `~/.guard/"$MAJOR_VER"` recommended by shellcheck?

Comment (umireon) [reply install-guard.sh:21]: This is my mistake. I'll fix them.

**Comment by umireon** [comment]
@priyap286 I've made the requested changes.

**Review by priyap286 — APPROVED** [review]

**Comment by priyap286** [comment]
> @priyap286 I've made the requested changes.

Thank you @umireon !

**Pull request merged**